src/new-storage/DeleteExpiredCmd.js

```diff
@@ -0,0 +1,134 @@
+const cassandra = require('cassandra-driver')
+const fetch = require('node-fetch')
+const pLimit = require('p-limit')
+
+const validateConfig = require('../helpers/validateConfig')
+
+class DeleteExpiredCmd {
+    constructor(config) {
+        validateConfig(config)
+
+        this.baseUrl = config.streamrUrl
+
+        const authProvider = new cassandra.auth.PlainTextAuthProvider(config.cassandraNew.username, config.cassandraNew.password)
+        this.cassandraClient = new cassandra.Client({
+            contactPoints: [...config.cassandraNew.hosts],
+            localDataCenter: config.cassandraNew.datacenter,
+            keyspace: config.cassandraNew.keyspace,
+            authProvider,
+            pooling: {
+                maxRequestsPerConnection: 32768
+            }
+        })
+
+        // used for limited concurrency
+        this.limit = pLimit(5)
+    }
+
+    async _getStreams() {
+        const result = []
+
+        const query = 'SELECT DISTINCT stream_id, partition FROM bucket'
+        const resultSet = await this.cassandraClient.execute(query).catch((err) => console.error(err))
+
+        if (resultSet.rows.length) {
```

Comment:
if `this.cassandraClient.execute(query)` errors, this will fail something like "cannot read value rows of undefined"

Reply:
true